### PR TITLE
Add more use cases in DG

### PR DESCRIPTION
**Pull request by yitong241**
Add more use cases in DG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -310,6 +310,44 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 (For all use cases below, the **System** is the `MedInfo` and the **Actor** is the `user`, unless specified otherwise)
 
+**Use case: UC01 - Add a patient**
+
```
Comment (nramapurath) [inline docs/DeveloperGuide.md:314]: Should we add the following line here before MSS:
Actor: User

**Review by nramapurath — COMMENTED** [review]
Good addition of the use case. I need to double check on the effects of the assertions, but we can sort it out in time while the use case is edited.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -310,6 +310,44 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**MSS**
+
+1. User request to add a new patient to the system.
+2. The user enters patient details.
+    1. The following are required information:
+        - Patient name
+        - Patient NRIC
+    2. The following are non-required information:
+        - Patient status
```
[the remark this answers is not in the document]
Comment (daytona65) [reply docs/DeveloperGuide.md:323]: Think is fine, i added ward under non-required as well

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -310,6 +310,44 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+   Use case ends.
+
+**Extensions**
+* 2a. If any of the required fields are not completed
+
+    * 2a1. the user is informed of this and show the correct format for the command
+
+Use case resume at step 2.
+
+* 2b. If the patient NRIC is duplicate when compared to the records in the system
```
[the remark this answers is not in the document]
Comment (daytona65) [reply docs/DeveloperGuide.md:336]: Done

**Pull request merged**